### ISSUE TITLE
CentOS -> Rocky?

### ISSUE THREAD

**Issue by karai17** [openresty/docker-openresty#207]
With CentOS 8 EOL, is there any consideration to migrate the CentOS 8 docker images over to Rocky Linux? Rocky is meant to be a drop-in replacement for CentOS and they also have official Docker images.

https://hub.docker.com/r/rockylinux/rockylinux

I bring this up because I am suddenly getting errors when trying to create a new docker image using `openresty/openresty:centos` as a base.

`Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist`

**Comment by zhuizhuhaomeng** [comment]
@neomantra  do you have any plan to upgrade CentOS to Rocky?

**Comment by karai17** [comment]
For postarity, I'd probaby archive `centos` and transition from centos 8.4 to rocky 8.5 and use the `rocky` tag going forward, though I am not exactly sure which version of centos 8 this docker image uses.

**Comment by neomantra** [comment]
Actually, not that my memory is refreshed, the different CentOS variants (like Amazon Linux) are handled with build flags on the `centos` Dockerfile, so the new flavor is not a new Dockerfile.  However, it looks like Rocky prefers `dnf` so the `fedora` Dockerfile works better.

This is what it looks like to DIY (change `RESTY_RPM_ARCH` to `amd64` if needed):
```
git clone https://github.com/openresty/docker-openresty.git

cd docker-openresty

docker build -f fedora/Dockerfile \
  -t openresty:rocky-8.5 \
  --build-arg RESTY_IMAGE_BASE=rockylinux \
  --build-arg RESTY_IMAGE_TAG=8.5 \
  --build-arg RESTY_YUM_REPO="https://openresty.org/package/rocky/openresty.repo" \
  --build-arg RESTY_RPM_DIST="el8" \
  --build-arg RESTY_RPM_ARCH="aarch64" \
  .
```

I will add this flavor and push to CI/CD today.

**Comment by karai17** [comment]
Cool, thanks! Yeah it looks like `dnf` is meant to finally replace `yum` with RHEL 8, though I'm sure it'll take until RHEL 9 before `yum` is deprecated and who knows when it will be removed.

One of the big differences between the variants is that they host their own software repos. openresty:centos correctly updated at some point to use the centos vault, but rocky et al host repos with continued support a la RHEL. I was getting the error in my OP because i had not rebuilt my base image before extending it and so I was trying to pull from the centos repos that no longer existed. rebuilding my image fixed that, but being stuck on software from 2021-12-31 is certainly not ideal.

**Comment by karai17** [comment]
Aye, rebuilding fixed my original issue. My stale image was trying to pull from the non-vaulted repo which no longer exists.

**Comment by karai17** [comment]
RE: using fedora instead of centos as the base

Why not use the actual rockylinux dockerfile as the base instead of fedora?

**Comment by neomantra** [comment]
Available now as image: `openresty/openresty:rocky`.  Will do a tag over the weekend.

**Comment by neomantra** [comment]
This is released now: `openresty/openresty:1.21.4.1-1-rocky`.
